src/server/database.ts

```diff
@@ -30,6 +30,27 @@ export interface AuthorTokenLive {
 export type AuthorToken = AuthorTokenMissing | AuthorTokenRefreshing
   | AuthorTokenValid | AuthorTokenLive;
 
+const getAuthorToken = async (
+  authorId: string,
+): Promise<AuthorTokenMissing | AuthorTokenRefreshing | AuthorTokenValid> => (
+  await get(`permanent:${authorId}`) || {
+    status: 'missing',
+  }
+);
+
+const setAuthorToken = async (
+  authorId: string,
+  authorToken: AuthorTokenRefreshing | AuthorTokenValid,
```

Comment:
I intentionally excluded `AuthorTokenMissing`, because we shouldn't ever try to store that information, and that allows TypeScript to enforce that at compile time. I'm certainly receptive to feedback that that's confusing or putting the responsibility for knowing such things in the wrong place, though!